File: backend/app/models/quiz.py
from typing import List, Optional
from .base import BaseEntity, CamelModel
from enum import Enum


class QuizType(str, Enum):
    MULTIPLE_CHOICE = "multiple_choice"
    TRUE_FALSE = "true_false"
    FILL_IN_THE_BLANK = "fill_in_the_blank"


class Quiz(BaseEntity):
    id: int
    word_id: int
    quiz_type: QuizType
    question: str
    options: List[str]
    correct_options: List[str]


class QuizCreate(CamelModel):
    word_id: Optional[int] = None
    quiz_type: QuizType
    question: str
    options: List[str]
    correct_options: List[str]


class QuizRequest(CamelModel):
    word_ids: List[int]

File: frontend/src/types/quiz.ts
export type QuizType = "multiple_choice" | "true_false" | "fill_in_the_blank";
 
export type Quiz = {
  id: number;
  createdAt: string;
  updatedAt: string;
  wordId: number;
  quizType: QuizType;
  question: string;
  options: string[];
  correctOptions: string[];
  quizStats: QuizStats;
};

export type QuizRequest = {
  wordIds: number[];
};

export type QuizStats = {
  currentIndex: number;
  score: number;
  selectedAnswer: string;
  startTime: number;
  correctAnswers: number;
  totalTime: number;
  answerResults: Record<number, boolean>;
}